backend/dataall/modules/shares_base/services/share_object_service.py

```diff
@@ -454,3 +688,19 @@ def _validate_group_membership(session, share_object_group, environment_uri):
                 action=CREATE_SHARE_OBJECT,
                 message=f'Team: {share_object_group} is not a member of the environment {environment_uri}',
             )
+
+    @staticmethod
+    def calculate_expiry_date(expirationPeriod, expirySetting):
+        currentDate = date.today()
```

Comment:
Here `date.today()` is used, but it can be called for existing shares as well.
So, there can be a case
1. Share is about to expire in 2 days
2. We update dataset's share expiration settings
3. We call  calculate_expiry_date
4. With `currentDate = date.today()` share is automatically prolonged for at least a month.

Reply:
Okay. Let me know if this is the correct scenario you are thinking,

1. Share is already approved with an expiration date
2. Someone modifies the Dataset and changes the expiration settings 
3. Then all shares are updated and thus this share will have a new expiration. 

If this is case you are thinking, I will be addressing this as a part of this comment - https://github.com/data-dot-all/dataall/pull/1489#discussion_r1727638952